logger/logger.go

```diff
@@ -51,44 +55,59 @@ type Message struct {
 	Options Options
 }
 
-func checkParameterLength(optionsType string, parameter string) {
-	if len(parameter) > 32 {
-		fmt.Println(optionsType + " length must be less than 32!")
-		os.Exit(3)
-	}
+// InvalidOptionMessage represents an issue with the supplied configuration.
+type InvalidOptionMessage struct {
+	Option  string
+	Message string
+}
+
+func (e InvalidOptionMessage) String() string {
+	return fmt.Sprintf("Options.%s: %s", e.Option, e.Message)
 }
 
-func checkOptions(options *Options) {
+func (options *Options) validate() error {
+	var problems []string
+	reMacAddress := regexp.MustCompile(`^([0-9a-fA-F][0-9a-fA-F]:){5}([0-9a-fA-F][0-9a-fA-F])`)
```

Comment:
any reason not to use https://golang.org/pkg/net/#ParseMAC?

Reply:
I considered that here, additional changes for validation to come in a separate issue